grammar.js

```diff
@@ -25,14 +25,16 @@ module.exports = grammar({
     null: $ => /null|NULL/,
     na: $ => /na|NA/,
     boolean: $ => /true|TRUE|false|FALSE/,
-    _number: $ => choice(
-      $.hex,
-      $.float,
-      $.integer,
-    ),
-    integer: $ => /\d+/,
+    _number: $ => seq(
+      optional("-"), 
+      choice(
+        $.hex,
+        $.float,
+        $.integer,
+    )),
```

Comment:
This expression doesn't need to change. Both `$.float` and `$.integer` already handle the optional `-`.